README.md

```diff
@@ -32,83 +53,34 @@ The common thread is that the use of compression dictionaries had run into vario
 
 A few things about this current proposal are different from past attempts, in ways we're hoping are meaningful:
 * CORS-based restrictions can ensure that public and private resources don't get mixed in ways that can leak user data.
-  - Dictionaries will be fetched with "omit" CORS mode to ensure they are public resources and don't contain any user secrets
-  - Non-shared dictionaries must be CORS fetched, and only apply to same destination
 * Path-based scoping would help us manage a "single possible dictionary per request" policy, which will minimize client-side cache fan-out.
+* Dictionaries must already be available on the client to be used (fetching of the dictionary is not in the critical path of a resource fetch).
 * Diff-caching on the server can simplify and enable the server-side deployment story.
 
 ## Use cases
 
-
 ### Compression types
 
-As mentioned above, use of compression dictionaries in browsers can be thought of as two very distinct categories:
+There are two primary models for using shared dictionaries that are similar but differ in how the dictionary is fetched:
 
 * Delta compression - reusing past downloaded resources for compressing future updates of the same or similar resources.
 * Shared dictionary - a dedicated dictionary is downloaded out-of-band, and then used to compress and decompress resources on the page.
 
-There are pros and cons to either of those approaches:
-
-
-* Shared dictionaries 
-    * Are likely to provide better compression ratios for resources never before seen by the user.
-    * They are easy to manage on the server side
-        * Only a small set of dictionaries to use for compression purposes.
-        * Precompression based on dictionaries users are likely to have is feasible.
-    * OTOH, they require out-of-band download which may only be worth the user’s while if they plan to visit this origin again. Predicting the future is hard..
-* Delta compression using past resources 
-    * Likely to provide better compression ratios for resource updates. 
-    * Doesn’t require out-of-band download of the dictionary.
-    * OTOH, server-side management can be hard:
-        * Servers may need to keep a large number of past versions to account for any user/past-resource combination
-        * As such, precompression may not be practical.
-
-
-### Resource types
-
-Otherwise, we have multiple categories of resources that can benefit from compression dictionaries.
-
-
-#### Static resources
+In both cases the client advertises the best-available dictionary that it has for a given request and if the server has a delta-compressed version of the resource using the same dictionary or has the dictionary available for dynamic compression then the dictionary is used as an external dictionary for the brotli-compressed response.
```

Comment:
Too long of a sentence. maybe 
"In both cases the client advertises the best-available dictionary that it has for a given request. If the server has a delta-compressed version of the resource, compressed with the advertized dictionary, it can just send that delta-compressed diff. It can also use that advertized dictionary (if available) to dynamically compress that resource." or something?

Reply:
Thanks, your version is great.